### ISSUE TITLE
Dropping items instantly disappear when using ark-server-tools

### ISSUE THREAD

**Issue by Q11x** [FezVrasta/ark-server-tools#81]
With current master branch and Ark version 180.4 and Linux ark server version 674165
When starting the server with ark-server-tools, items will instantly disappear in-game when dropping them.
When starting the server up without using ark-server-tools the issue isn't present.

Other people is having the same problem as can be seen here:
http://steamcommunity.com/app/346110/discussions/0/523890046878824192/ 

### Resolution
* **Never** define the same settings in both arkmanager.cfg AND GameUserSettings.ini. Stick with one of them.
* In your arkmanager.cfg, remove the line that includes servermail=" "

**Comment by Atriusftw** [comment]
Hm, weird.
I just tested this on my server, running latest version from master branch and build 674165 (180.4)

[My loot shows up](http://atrius.net/~random/2015-06-28_00001.jpg) just fine.

**Comment by Q11x** [comment]
I didn't have the problem before updating. But as I updated both the script and server at the same time, I don't know which of the two that made the affecting change.

**Comment by klightspeed** [comment]
@Q11x Do you have any custom ark_* options in /etc/arkmanager/arkmanager.cfg or ~/.arkmanager.cfg?

**Comment by PhuriousGeorge** [comment]
Is there an output to a log of the full startup command arktools is passing? I'm running into the same issue.

**Comment by Q11x** [comment]
This is silly but I fixed it. 
By setting
servermail="some@mail.adress"
back to
servermail=""
in the arkmanger config
it started working again.

**Comment by Atriusftw** [comment]
Right, might aswell remove that from the config. That got removed from arkmanager some commits ago.

**Comment by Q11x** [comment]
I'll wait for @PhuriousGeorge to comment to see if that fixed it for him as well before I close the issue.

**Comment by klightspeed** [comment]
@Q11x @Atriusftw How that would be causing the problem, I don't know, since it's never referenced by the script and is not exported to the server's environment.

Did you do anything else, like changing to another directory?

**Comment by Q11x** [comment]
@klightspeed That's the only thing I did to fix it. 
Stopped the server by "Ctrl + c", that I earlier had started up with: (Where the problem wasn't present)
```
./ShooterGameServer TheIsland?listen?SessionName=somename?ServerPassword=somepassword?ServerAdminPassword=somepassword-server -log
```
Took a backup of the world save.
Then changed:
`servermail="some@mail.adress"`
back to
`servermail=""`
in the config.
Started the server back up with:
`arkmanager start`
And the problem was gone. 

Everything have been done with the same user.

**Current arkmanager config:**
```
# config SteamCMD
steamcmdroot="/home/steam/steamcmd"                                 # path of your steamcmd instance
steamcmdexec="steamcmd.sh"                                          # name of steamcmd executable
steamcmd_user="steam"                                               # name of the system user who own steamcmd folder
steamcmd_appinfocache="/home/steam/Steam/appcache/appinfo.vdf"      # cache of the appinfo command

# config environment
arkserverroot="/home/steam/steamapps/ark"                                     # path of your ARK server files (default ~/ARK)
arkserverexec="ShooterGame/Binaries/Linux/ShooterGameServer"        # name of ARK server executable

# ARK server options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap="TheIsland"                                               # server map (default TheIsland)
ark_SessionName="removed"                                  # if your session name needs special characters please use the .ini instead
ark_Port="7777"                                                     # ARK server port (default 7778)
ark_QueryPort="27015"                                               # ARK query port (default 27016)
ark_ServerPassword="removed"                                               # ARK server password, empty: no password required to login
ark_ServerAdminPassword="removed"                               # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers="22"

# config Service
servicename="arkserv"                                               # Name of the service (don't change if you don't know what are you doing)
logdir="/var/log/arktools"                                          # Logs path (default /var/log/arktools)

# steamdb specific
appid=376030                                                        # Linux server App ID

# admin information
servermail=""                                                       # Log email, leave blank if you dont want to receive mail

```

**Comment by Atriusftw** [comment]
Just for the record, i never defined an email adress either.

**Comment by klightspeed** [comment]
That first command suggests that you `cd /home/steam/steamapps/ark/ShooterGame/Binaries/Linux`

Then backing up the world save suggests that you `cd /home/steam/steamapps/ark/ShooterGame/Saved/SavedArks`

Where did you `cd` when it wasn't working?

In any case, I think we should set the current working directory to e.g. the ARK root directory before starting.

**Comment by Atriusftw** [comment]
Confirmed by players in the steam discussion thread that removing the servermail from config resolves the problem.

:white_check_mark: Hotfix merged into master. Closing issue =)

**Comment by PhuriousGeorge** [comment]
Just applied the update, still seeing an issue here.

Just fixed it - wasn't commenting out duplicate variables:

arkmanager.cfg:
'''
# ARK server options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap="TheIsland"                                               # server map (default TheIsland)
ark_SessionName="****"                                 # if your session name needs special characters please use the .ini instead
ark_Port="7778"                                                     # ARK server port (default 7778)
ark_QueryPort="27016"                                               # ARK query port (default 27016)
ark_ServerPassword=""                                               # ARK server password, empty: no password required to login
ark_ServerAdminPassword="****"                                 # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers="100"
'''

**Comment by Atriusftw** [comment]
You should **never** define arguments in two places.

**Comment by PhuriousGeorge** [comment]
Yup, an argument got added and I missed commenting it out. I'll watch more closely now.

**Comment by Atriusftw** [comment]
I think this issue is back.

I'm running 1.3-dev and the items dissapear instantly on dropping them.
Can't seem to figure out what it might be.

The server logs shows the following though;

``` 
/usr/local/bin/arkmanager: line 278:  4058 Aborted                 "$arkserverroot/$arkserverexec" "$arkserveropts"
```

**Comment by klightspeed** [comment]
The `Aborted` error message means that the server crashed.

To exclude the previously attributed cause, please try prepending line 278 with `env -i `, and let us know if the problem still occurs.  `env -i

**Comment by Atriusftw** [comment]
I'm getting 4923 Killed aswell. Though these errors only get logged when i manually stop the server with either doexit or quit in the in-game console. Don't think it's a problem then.

I'm actually more worried about the changes that happens to GameUserSettings.ini when starting the server.

**Comment by Lihis** [comment]
I experienced this same issue, even when I did not have arkmanager
installed. But this solution worked for me at the post #5
http://steamcommunity.com.tw/app/346110/discussions/0/523890046883024363/#c535150948594431775

**Comment by Atriusftw** [comment]
Thank you for the information, @Lihis. I will forward that to the ark devs.

**Comment by tenten36** [comment]
I get a big red warning from the link @Lihis provided in Opera, seems to be a phishing site [.com.tw]. Can you remove it?

**Comment by Lihis** [comment]
@tenten36 I did not notice that on Chrome so thanks for the info! Now the link should point to the right steamcommunity URL.

**Comment by tenten36** [comment]
I didn't even know Opera has that feature...thanks dude

**Comment by Lihis** [comment]
@Atriusftw My server was running the latest 186.3 version.

**Comment by Atriusftw** [comment]
@Lihis Thanks. Did you have this problem prior to updating your server version?

**Comment by Lihis** [comment]
@Atriusftw This was the first time when I installed the ARK server and I got that version installed.

**Comment by Atriusftw** [comment]
**Resolved in patch 187**
